src/nfa_builder.rs

```diff
@@ -231,37 +230,26 @@ where
         // But, there is no problem since Daachorse does not allow an empty pattern.
         debug_assert_ne!(q[0], ROOT_STATE_ID);
 
+        // Adds a dummy output so that the output_pos is positive.
+        self.outputs.push(Output::new(0, 0, None));
+
         for &state_id in q {
             let s = &mut self.states[state_id as usize].borrow_mut();
             if s.output.0 == VALUE_INVALID {
                 s.output_pos = self.states[s.fail as usize].borrow().output_pos;
                 continue;
             }
 
-            s.output_pos = self.outputs.len().try_into().unwrap();
+            s.output_pos = NonZeroU32::new(self.outputs.len().try_into().unwrap());
             let parent = self.states[s.fail as usize].borrow().output_pos;
 
             self.outputs
                 .push(Output::new(s.output.0, s.output.1, parent));
-            Self::check_outputs_error(&self.outputs)?;
```

Comment:
> Is this because the same error handling is done where this function is called?

Yes. Furthermore, the current implementation uses the same threshold for both charwise and bytewise automata.

> Moreover, I guess this function always returns `Ok(())`. So, if this error handling is removed, I think the return type of the function should be `()`.

I fix the return type in #34

Reply:
I understood, thanks!